lib/src/message_window.dart

```diff
@@ -12,32 +14,75 @@ final _emptyData = Uint8List(0);
 
 typedef _ServerMessageFn = ServerMessage Function(Uint8List data);
 
-Map<int, _ServerMessageFn> _messageTypeMap = {
-  49: (d) => ParseCompleteMessage(),
-  50: (d) => BindCompleteMessage(),
-  65: NotificationResponseMessage.new,
-  67: CommandCompleteMessage.new,
-  68: DataRowMessage.new,
-  69: ErrorResponseMessage.new,
-  75: BackendKeyMessage.new,
-  82: AuthenticationMessage.new,
-  83: ParameterStatusMessage.new,
-  84: RowDescriptionMessage.new,
-  87: CopyBothResponseMessage.new,
-  90: ReadyForQueryMessage.new,
-  100: CopyDataMessage.new,
-  110: (d) => NoDataMessage(),
-  116: ParameterDescriptionMessage.new,
-  $3: (d) => CloseCompleteMessage(),
-};
+// Map<int, _ServerMessageFn> _messageTypeMap = {
+//   49: (d) => ParseCompleteMessage(),
+//   50: (d) => BindCompleteMessage(),
+//   65: NotificationResponseMessage.new,
+//   67: CommandCompleteMessage.new,
+//   68: DataRowMessage.new,
+//   69: ErrorResponseMessage.new,
+//   75: BackendKeyMessage.new,
+//   82: AuthenticationMessage.new,
+//   83: ParameterStatusMessage.new,
+//   84: RowDescriptionMessage.new,
+//   87: CopyBothResponseMessage.new,
+//   90: ReadyForQueryMessage.new,
+//   100: CopyDataMessage.new,
+//   110: (d) => NoDataMessage(),
+//   116: ParameterDescriptionMessage.new,
+//   $3: (d) => CloseCompleteMessage(),
+// };
+
+_ServerMessageFn _messageTypeMap(int? msgType, Encoding encoding) {
```

Comment:
This is not used outside of `MessageFramer`, right? Maybe we can move the method there now, that avoids the `encoding` as an extra parameter since it's already a field there. We can also avoid the intermediate function object as well then:

```dart
  ServerMessage _readMessage(int? msgType, Uint8List data) {
    switch (msgType) {
      case 49:
        return ParseCompleteMessage();
     // ...
     case 68:
       return DataRowMessage(data);
    // ...
     default:
       return UnknownMessage(msgType, data);
    }
  }
```

Reply:
done